src/ddmd/func.d

```diff
@@ -5294,9 +5294,18 @@ extern (C++) final class InvariantDeclaration : FuncDeclaration
  */
 extern (C++) static Identifier unitTestId(Loc loc)
 {
+    // there's a counter per module in case there are two unit tests on one line
+    static __gshared int[const(char)*] counterPerModule;
```

Comment:
I very much doubt that any discernible difference will exist in performance due to naming unittest blocks. Delaying where the function name is set seems to be quite involved work - I'm not sure it's worth it. Especially since we're talking about how unittests are named here.

Of course, if needs be...